File: src/models/connector.ts
export interface Connector {
  id: string;
  name: string;
  encrypted_config: string;
  created_at: string;
  updated_at: string;
}

export interface CreateConnectorData {
  name: string;
  encrypted_config: string;
}

export interface UpdateConnectorData {
  encrypted_config?: string;
}

export type ConnectorName = 'plaid' | 'cherry' | 'docusign' | 'dropboxsign' | 'clear' | string;

File: server/models/connector.py
"""Connector model for external service integrations."""

from sqlalchemy import Column, String, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime

from .base import Base


class Connector(Base):
    """External service connector configurations (encrypted, tenant-scoped)."""
    __tablename__ = "connectors"
    
    tenant_id = Column(String, ForeignKey("tenants.id"), primary_key=True, nullable=False, index=True)
    name = Column(String, primary_key=True, nullable=False)  # docusign, plaid, clear, cherry_sms, dropbox_sign
    encrypted_config = Column(Text, nullable=False)  # Fernet encrypted JSON
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    tenant = relationship("Tenant", backref="connectors")